api/src/opentrons/protocol_api/module_contexts.py

```diff
@@ -770,3 +784,194 @@ def total_step_count(self) -> Optional[int]:
+        hw_module: modules.heater_shaker.HeaterShaker,
+        geometry: HeaterShakerGeometry,
+        requested_as: ModuleModel,
+        at_version: APIVersion,
+        loop: asyncio.AbstractEventLoop,
+    ) -> None:
+        self._module = hw_module
+        self._loop = loop
+        super().__init__(ctx, geometry, requested_as, at_version)
+
+    # TODO: add API version requirement
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Target temperature of the heater-shaker's plate."""
+        return self._module.target_temperature
+
+    # TODO: add API version requirement
+    @property
+    def current_temperature(self) -> float:
+        """Current temperature of the heater-shaker's plate."""
+        return self._module.temperature
+
+    # TODO: add API version requirement
+    @property
+    def current_speed(self) -> int:
+        """Current speed of the heater-shaker's plate."""
+        return self._module.speed
+
+    # TODO: add API version requirement
+    @property
+    def target_speed(self) -> Optional[int]:
+        """Target speed of the heater-shaker's plate."""
+        return self._module.target_speed
+
+    # TODO: add API version requirement
+    @property
+    def temperature_status(self) -> str:
+        """Heater-shaker's temperature status string.
+
+        Returns one of these possible status values:
+        - "holding at target"
+        - "cooling"
+        - "heating"
+        - "idle"
+        - "error"
+        """
+        return self._module.temperature_status.value
+
+    # TODO: add API version requirement
+    @property
+    def speed_status(self) -> str:
+        """Heater-shaker's speed status string.
+
+        Returns one of these possible status values:
+        - "holding at target"
+        - "speeding up"
+        - "slowing down"
+        - "idle"
+        - "error"
+        """
+        return self._module.speed_status.value
+
+    # TODO: add API version requirement
+    @property
+    def labware_latch_status(self) -> str:
+        """Heater-shaker's labware latch status string.
+
+        Returns one of these possible status values:
+        - "opening": latch is opening
+        - "idle_open": latch is open and idle
+        - "closing": latch is closing
+        - "idle_closed": latch is closed and idle
+        - "idle_unknown": status upon reset
+        - "unknown": latch status cannot be reached, likely due to an error
+        """
+        return self._module.labware_latch_status.value
+
+    # TODO: add API version requirement
+    def set_and_wait_for_temperature(self, celsius: float) -> None:
+        """Set and wait for target temperature.
+
+        Sets the heater-shaker target temperature and delays protocol execution
+        until target temperature has reached.
+
+        Note: The H/S truncates the temperature param to 2 decimal places
+
+        :param celsius: The target temperature, in °C in range 37°C to 95°C.
+        """
```

Comment:
```suggestion
        """Set the target temperature and wait for it to be reached.

        Note: The Heater-Shaker truncates the ``temperature`` parameter to 2 decimal places.

        :param celsius: The target temperature, in °C in range 37°C to 95°C.
        """
```

* Combined 1st and 2nd sentences, for brevity.
* Spelled `H/S` out as `Heater-Shaker`, spelled out `parameter`, and added a period.

Reply:
@SyntaxColoring sorry it wasn't clear that these docstrings may all be obviated by #10686. [There](https://github.com/Opentrons/opentrons/pull/10686/files#diff-1e43020f8a96546c84d849654ddbd737240a92e3c3724becc7aaa83d76fde041R883-R885) I have 
```
Set a target temperature and block execution of further commands until the module reaches the target. 

        :param celsius: A value between 27 and 95, representing the target temperature in °C. Values are automatically truncated to two decimal places, and the Heater-Shaker module has a temperature accuracy of ±0.5 °C.
```